### DISCUSSION TITLE
Linux Binary not as portable as advertised

### DISCUSSION THREAD

**Discussion by rakzcs** [ImageMagick/ImageMagick#5526] [Help] [answered]
Just downloaded and it wont run.
```
dlopen(): error loading libfuse.so.2

AppImages require FUSE to run. 
You might still be able to extract the contents of this AppImage 
if you run it with the --appimage-extract option. 
See https://github.com/AppImage/AppImageKit/wiki/FUSE 
for more information
```

Also even when installing FUSE there is another error

```
fuse: device not found, try 'modprobe fuse' first

Cannot mount AppImage, please check your FUSE setup.
You might still be able to extract the contents of this AppImage 
if you run it with the --appimage-extract option. 
See https://github.com/AppImage/AppImageKit/wiki/FUSE 
for more information
open dir error: No such file or directory
```

**Comment by urban-warrior** [comment, marked as answer]
ImageMagick is portable as advertised assuming you have a proper AppImage environment.  AppImage itself has a Fuse requirement, not ImageMagick.   See https://appimage.org/ for AppImage installation help.

**Comment by rakzcs** [comment]
Any good guides on how to build from source a command line only imagemagick binary?

**Reply by urban-warrior** [reply]
https://imagemagick.org/script/install-source.php

**Reply by rakzcs** [reply]
After the build, where are the executable binaries. The problem i'm having is that I need to move the binaries from one machine to the other.

**Reply by urban-warrior** [reply]
Did you review https://imagemagick.org/script/advanced-linux-installation.php?  You likely wand a static "uninstalled" version of ImageMagick.  The binaries would land in the `utilities` folder.

**Reply by rakzcs** [reply]
I did not. Now i created a configuration with `./configure --disable-installed --disable-shared`
The only issue that i'm having at the moment is that `error while loading shared libraries: libgomp.so.1` Its possible to install it, but would there be a possibility to bundle this?

**Reply by urban-warrior** [reply]
You can add `--disable-openmp` to your configure command-line to eliminate the dependency but that would disable threading and slow down ImageMagick on multi-core hosts.  You might be able to install a static libgomp on your build host to statically link libgomp to your ImageMagick binary.

**Reply by rakzcs** [reply]
Next issue, Delegates options wont enable on ./configure
Tried multiple different variants, but always end up with the default zlib

**Reply by urban-warrior** [reply]
Not sure what you're asking.  Do you want to point to another Zlib install?  Use `CFLAGS=/path LDFLAGS=/path ./configure  arguments`.

**Reply by rakzcs** [reply]
Trying to enable some of these delegates
```Delegate library configuration:
  BZLIB             --with-bzlib=yes		no
  Autotrace         --with-autotrace=no		no
  DJVU              --with-djvu=yes		no
  DPS               --with-dps=yes		no
  FFTW              --with-fftw=no		no
  FLIF              --with-flif=no		no
  FlashPIX          --with-fpx=yes		no
  FontConfig        --with-fontconfig=yes	no
  FreeType          --with-freetype=yes		no
  Ghostscript lib   --with-gslib=yes		no
  Graphviz          --with-gvc=yes		no
  HEIC              --with-heic=yes		no
  JBIG              --with-jbig=yes		no
  JPEG v1           --with-jpeg=yes		no
  JPEG XL           --with-jxl=no		no
  LCMS              --with-lcms=yes		no
  LQR               --with-lqr=yes		no
  LTDL              --with-ltdl=no		no
  LZMA              --with-lzma=yes		no
  Magick++          --with-magick-plus-plus=yes	yes
  OpenEXR           --with-openexr=yes		no
  OpenJP2           --with-openjp2=yes		no
  PANGO             --with-pango=yes		no
  PERL              --with-perl=no		no
  PNG               --with-png=yes		no
  RAQM              --with-raqm=yes		no
  RAW               --with-raw=yes		no
  RSVG              --with-rsvg=yes		no
  TIFF              --with-tiff=yes		no
  WEBP              --with-webp=yes		no
  WMF               --with-wmf=yes		no
  X11               --with-x=yes			no
  XML               --with-xml=yes		no
  ZIP               --with-zip=yes		no
  ZLIB              --with-zlib=yes		yes
  ZSTD              --with-zstd=yes		no
```

**Reply by urban-warrior** [reply]
Take FFTW, for example.  Install the development version of the delegate library on your build host.  Then add `--with-fftw` to your `configure` command-line.  If the validation fails, consult `config.log` for the reason, fix the issue, and try again.

**Reply by PasqualePuzio** [reply]
> Trying to enable some of these delegates
> 
> ```
>   BZLIB             --with-bzlib=yes		no
>   Autotrace         --with-autotrace=no		no
>   DJVU              --with-djvu=yes		no
>   DPS               --with-dps=yes		no
>   FFTW              --with-fftw=no		no
>   FLIF              --with-flif=no		no
>   FlashPIX          --with-fpx=yes		no
>   FontConfig        --with-fontconfig=yes	no
>   FreeType          --with-freetype=yes		no
>   Ghostscript lib   --with-gslib=yes		no
>   Graphviz          --with-gvc=yes		no
>   HEIC              --with-heic=yes		no
>   JBIG              --with-jbig=yes		no
>   JPEG v1           --with-jpeg=yes		no
>   JPEG XL           --with-jxl=no		no
>   LCMS              --with-lcms=yes		no
>   LQR               --with-lqr=yes		no
>   LTDL              --with-ltdl=no		no
>   LZMA              --with-lzma=yes		no
>   Magick++          --with-magick-plus-plus=yes	yes
>   OpenEXR           --with-openexr=yes		no
>   OpenJP2           --with-openjp2=yes		no
>   PANGO             --with-pango=yes		no
>   PERL              --with-perl=no		no
>   PNG               --with-png=yes		no
>   RAQM              --with-raqm=yes		no
>   RAW               --with-raw=yes		no
>   RSVG              --with-rsvg=yes		no
>   TIFF              --with-tiff=yes		no
>   WEBP              --with-webp=yes		no
>   WMF               --with-wmf=yes		no
>   X11               --with-x=yes			no
>   XML               --with-xml=yes		no
>   ZIP               --with-zip=yes		no
>   ZLIB              --with-zlib=yes		yes
>   ZSTD              --with-zstd=yes		no
> ```

Hi @rakzcs did you manage to build a portable version of imagemagick? I have the same issue with delegates, what libraries did you have to install?

**Reply by rakzcs** [reply]
Hi

Yes I did get it to work. This thread here was really helpful and solved the issues https://gist.github.com/bensie/56f51bc33d4a55e2fc9a

**Reply by PasqualePuzio** [reply]
Thank  you @rakzcs I did manage to fix that issue too and create the executables. However it turns out they're not statically linked as I explained here https://github.com/ImageMagick/ImageMagick/discussions/5996
Do you have the same issue? Any idea of how to make it statically linked?

**Reply by rakzcs** [reply]
No clue on how to do it, maybe @urban-warrior has some insight into it.

**Reply by hsf4e** [reply]
Summary of problems we experience.
1. In fact ImageMagic has indirect dependency to AppImage environment.
2. AppImage requires FUSE to be installed.
3. Installation of FUSE requires installing kernel module, so 
4. So when we have server with limited access, we cannot install "portable" program for image conversion because it needs installing new kernel module, so we need SU access!

Bonus 1: 
Windows binaries are 26! + MinGW + CygWin = 28
Linux "portable" non-distro specific binaries: 1 (with new kernel module dependencies)

Bonus 2: the "portable" binary is promoted as "Complete portable application on Linux, no installation required. Just download and run."
